backend/general_functions.py

```diff
@@ -0,0 +1,372 @@
+        raise Exception('cannot set alert to this level')
+
+    create('alert', data)
+
+# 13- list of all regions
+def get_regions():
+    rows = get_all('region')
+    for region in rows:
+        cities = get_single('city', {'regionName' : region['name']})
+
+        for city in cities:
+            postal_codes = get_single('cityPostalCodeMapping', {'city' : city['name']})
+            temp = list(map(lambda x: x['postalCodeRegion'], postal_codes))
+            city['postalCodes'] = temp
+            city.pop('regionName', None)
+
+        region['cities'] = cities
+        region.pop('alertLevel', None)
+
+    return rows
+
+# 14- list of people who got the result of the test on a specific date
+def get_test_result_on_date(data):
+    conn = get_conn()
+    cursor = conn.cursor()
+    
+    query = ("""
+    SELECT P.firstName, P.lastName, P.dateOfBirth, P.phoneNumber, P.email, D.result
+FROM
+(SELECT medicareNumber, result
+FROM diagnostic
+WHERE testDate = '{}') as D, person as P
+WHERE P.medicareNumber = D.medicareNumber
+ORDER BY D.result DESC
+;""").format(data['testDate'])
+
+    cursor.execute(query)
+
+    columns = [col[0] for col in cursor.description]
+    rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+
+    cursor.close()
+    conn.commit()
+    conn.close()
+
+    return rows
+
+# 15- list of workers in specific facility
+def get_workers_at_facility(data):
+    conn = get_conn()
+    cursor = conn.cursor()
+
+    query = ("""SELECT P.firstName, P.lastName, PHW.workerID, PHW.medicareNumber, PHW.publicHealthCenterID
+FROM publicHealthWorker as PHW, person as P
+WHERE publicHealthCenterID = {} and PHW.medicareNumber = P.medicareNumber;""").format(data['publicHealthCenterID'])
+
+    cursor.execute(query)
+
+    columns = [col[0] for col in cursor.description]
+    rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+
+    cursor.close()
+    conn.commit()
+    conn.close()
+
+    return rows
+
+# 16- list of all public health workers who tested positive on a specific date in a specific facility
+def get_all_at_risk_workers(from_date, to_date, workerid):
+    conn = get_conn()
+    cursor = conn.cursor()
+
+    query = ("""SELECT DISTINCT PHW2.workerID, P.firstName, P.lastName
+FROM workerSchedule, publicHealthWorker as PHW1, publicHealthWorker as PHW2, person as P
+WHERE 
+	workerSchedule.workerID <> {} and 
+    `date` between "{}" and "{}" and 
+    PHW1.workerID = {} and
+    PHW2.workerID = workerSchedule.workerID and
+    PHW1.publicHealthCenterID = PHW2.publicHealthCenterID and
+    PHW2.medicareNumber = P.medicareNumber;""").format(workerid, from_date, to_date, workerid)
+
+    cursor.execute(query)
+
+    columns = [col[0] for col in cursor.description]
+    rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+
+    cursor.close()
+    conn.commit()
+    conn.close()
+    return rows
+
+def get_workers_positive_test_at_facility(data):
```

Comment:
DB structure for workers and hospitals has changed, this will need to be updated